ledgers/ccf/transaction_processor/pdo_tp.cpp

```diff
@@ -54,26 +55,70 @@ namespace ccfapp
+            // get the current view of contract_enclave_check_attestation_flag
+            auto check_attestation_flag_view = ctx.tx.rw(contract_enclave_check_attestation_flag);
+
+            // Current PDO policy permits the flag to be set only once. Check if already set.
+
+            // Below we check the ccf node has a local copy of the attestation flag. If yes, an error
+            // is returned. (Note that global commit of the flag might be pending, and this is OK).
+            auto check_attestation_flag_check = check_attestation_flag_view->get(PDO_ENCLAVE_CHECK_ATTESTATION_FLAG);
+            if (check_attestation_flag_check.has_value()){
+                return ccf::make_error(
+                    HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput,"Attesation check flag can be set only once");
+            }
 
             // collect the data to be stored
-            ContractEnclaveAttestationVerificationPolicy attestation_policy;
+            ContractEnclaveAttestionCheckFlag check_attestation_flag;
+            check_attestation_flag.check_attestation = in.check_attestation;
+
+            //store the data
+            check_attestation_flag_view->put(PDO_ENCLAVE_CHECK_ATTESTATION_FLAG, check_attestation_flag);
 
-            attestation_policy.check_attestation = in.check_attestation;
-            if(in.check_attestation){
-                attestation_policy.mrenclave = in.mrenclave;
-                attestation_policy.basename = in.basename;
-                attestation_policy.ias_public_key = in.ias_public_key;
+            return ccf::make_success(true);
+        };
+
+        // register PDO enclave expected SGX measurements (member method)
+        // Note that this RPC may be called only setting the
+        // set_contract_enclave_attestation_check_flag to true.
+        auto set_contract_enclave_expected_sgx_measurements = [this](auto& ctx, const nlohmann::json& params) {
+
+            //ensure that attestation check flag is true; otherwise throw error
+
+            // get the current view of contract_enclave_check_attestation_flag
+            auto check_attestation_flag_view = ctx.tx.rw(contract_enclave_check_attestation_flag);
+            auto check_attestation_flag_global = check_attestation_flag_view->get_globally_committed(PDO_ENCLAVE_CHECK_ATTESTATION_FLAG);
+            if (!check_attestation_flag_global.has_value()){
+                return ccf::make_error(
+                    HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput,"Please enable attesation check before providing expected measurements");
+            }
+            auto check_attestation_flag = check_attestation_flag_global.value();
+            if (!check_attestation_flag.check_attestation){
+                return ccf::make_error(
+                    HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput,"Please enable attesation check before providing expected measurements");
```

Comment:
I don't think these checks are strictly necessary (since we are just setting, but not using, the measurements) but it helps getting error messages early on in the case of misconfigurations.

Reply:
thank you, IMHO, setting sgx measurements without having set the check flag to True is confusing, and best not permitted by the TP. retaining the code as is.